gwr/sel_bw.py

```diff
@@ -244,13 +257,9 @@ def _init_section(self, X_glob, X_loc, coords, constant):
             a = 40 + 2 * n_vars
             c = n
         else:
-            nn = 40 + 2 * n_vars
-            sq_dists = squareform(pdist(coords))
-            sort_dists = np.sort(sq_dists, axis=1)
-            min_dists = sort_dists[:,nn-1]
```

Comment:
We discussed changing this to just half the minimum and twice the maximum last call? That's why I did what I did. 

Currently, this will do half the maximum interpoint distance for the maximum, and half the minimum (nn-1)th neighbor distance. 

My change suggests half the minimum interpoint distance and twice the maximum interpoint distance. In general, this widens the search. 

I would also be fine with just doing twice the maximum (instead of half the maximum), and leaving the lower bound as is. But, we need to consider bigger bandwidths, since our current stuff is very clearly hitting the max bandwidth before its finding a minimum in some of the test cases.

Reply:
Yes, ok, I remember now. Didn't realize it was coming on this PR. Makes sense to me, and I like the idea of widening the search, especially given that we've optimized the routines, so even it requires a few more evaluations to find the optimal bandwidth all in all it shouldn't add much more computation time, but does eliminate some of the chance of initializing starting values that result in a sub-optimal solution.